### PR TITLE
Minor Refactoring

### PR DESCRIPTION
**Pull request by siddhi-gh**
### Description

- Install mui components package.
- Extract utility functions.
- Extract theme context.
- Refactor files structure.
- Style improvements

**NOTE:** This PR is introducing many changes, please test the App thoroughly and report if bugs are found.

### Before
Screenshot 2023-05-30 at 12 08 36 AM

### After
Screenshot 2023-05-30 at 12 08 45 AM

### Task Remaining
- Fix the Transition animation of `Tile` on click.

### CODE REVIEW

File: src/tiles/Tiles.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+  const handleClick = () => {
+    if (tileName === "Color Shuffle") {
+      shuffleColors(); // Call the shuffleColors function passed from App.js
+    } else {
+      setIsFlipped(!isFlipped);
+    }
+  };
+
+  const renderContent = () => {
```
Comment (siddhi-gh) [inline src/tiles/Tiles.js:45]: I may have to revisit this logic again.

File: src/tiles/Tiles.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    if (content.type === "iconLink") {
+      return (
+        <a href={content.url}>
+          <IconContainer>
+            {!isFlipped ? (
+              <FlippedTile src={content.iconSrc} alt={content.iconAlt} />
+            ) : (
+              <Tile />
```
Comment (siddhi-gh) [inline src/tiles/Tiles.js:53]: Not sure if this is a good rendering.

File: src/tiles/Tiles.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    } else {
+      return <TileContent>{content}</TileContent>;
+    }
+  };
+
+  return (
+    <Tile style={tileStyle} onClick={handleClick}>
+      <TileName>
+        {isFlipped ? <TileName>{tileName}</TileName> : renderContent()}
+      </TileName>
```
Comment (siddhi-gh) [inline src/tiles/Tiles.js:67]: May need to redo this too.

File: src/tiles/Tiles.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  tileName,
+  content,
+  isSquare,
+  color,
+  shuffleColors,
+}) => {
+  const isDarkMode = useContext(ThemeContext);
+  const [isFlipped, setIsFlipped] = useState(false);
+
+  const tileStyle = {
```
Comment (siddhi-gh) [inline src/tiles/Tiles.js:22]: I think this can be moves to the styles file.

File: src/tiles/Tiles.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    "--text-color": isDarkMode ? "white" : "black",
+  };
+
+  const tileSize = `${getRandomSize()}px`;
+
+  if (isSquare) {
+    tileStyle["--tile-width"] = tileSize;
+    tileStyle["--tile-height"] = tileSize;
+  } else {
+    tileStyle["--tile-width"] = "400px";
```
Comment (siddhi-gh) [inline src/tiles/Tiles.js:33]: Can be moved to styles.

**Review by siddhi-gh — COMMENTED** [review]
Points to note

File: src/app/App.js [hunk 6]
```diff
@@ -0,0 +1,66 @@
+import React, { useState, useEffect } from "react";
+import { Sun, Moon } from "react-feather";
+import {
```
Comment (tejas-k3) [inline src/app/App.js:3]: Weird line break

File: src/app/AppStyles.js
Comment (tejas-k3) [inline src/app/AppStyles.js:1]: If file is specific under app path, why styling is appstyles.js instead of just styles.js?

Comment (siddhi-gh) [reply src/app/AppStyles.js:1]: for better redability.

File: src/app/App.js (hunk 6 above)
Comment (siddhi-gh) [reply src/app/App.js:3]: Prettify extension did that.

Comment (siddhi-gh) [reply src/app/App.js:3]: and this is the general imports convention for large function imports from a single source.

File: src/tiles/Tiles.js (hunk 4 above)
Comment (tejas-k3) [reply src/tiles/Tiles.js:22]: The reason being?

File: src/tiles/Tiles.js (hunk 2 above)
Comment (tejas-k3) [reply src/tiles/Tiles.js:53]: Reference link/document for same?

File: src/tiles/Tiles.js (hunk 4 above)
Comment (siddhi-gh) [reply src/tiles/Tiles.js:22]: Because it basically sets the tike color which is a css attribute.

**Review by tejas-k3 — APPROVED** [review]

**Pull request merged**